packages/teams-js/src/private/messageChannels.ts

```diff
@@ -0,0 +1,59 @@
+import { requestPortFromParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * Namespace to request message ports from the host application.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace messageChannels {
+  let telemetryPort: MessagePort | undefined;
+  /**
+   * @hidden
+   *
+   * Fetches a MessagePort to batch telemetry through the host's telemetry worker.
+   * @returns MessagePort.
+   *
+   * @throws Error if {@linkcode app.initialize} has not successfully completed,
+   * if the host does not support the feature, or if an error is thrown in message handling.
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export async function getTelemetryPort(): Promise<MessagePort> {
+    // If the port has already been initialized, return it.
+    if (telemetryPort) {
```

Comment:
Is it safe to assume that the telemetry port never changes? Would be good to add a short log statement here saying that the cached telemetry port is being used. Would also be good to add that information to the docs above the function.

Reply:
Yes, for a single iFrame, the intent is to have a single port that is requested during start up. Ideally the iFrame should cache the port itself within its telemetry framework, but if for some reason it is rerequested, we don't want a new message channel to be created.